spanner/api/Spanner.Samples.Tests/UpdateDatabaseAsyncTest.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Cloud.Spanner.Admin.Database.V1;
+using Google.Cloud.Spanner.Common.V1;
+using Google.Protobuf.WellKnownTypes;
+using System.Threading.Tasks;
+using Xunit;
+
+[Collection(nameof(SpannerFixture))]
+public class UpdateDatabaseAsyncSampleTest
+{
+    private readonly SpannerFixture _spannerFixture;
+
+    public UpdateDatabaseAsyncSampleTest(SpannerFixture spannerFixture)
+    {
+        _spannerFixture = spannerFixture;
+    }
+
+    [Fact]
+    public async Task UpdateDatabaseAsyncSample()
+    {
+        var databaseAdminClient = DatabaseAdminClient.Create();
+
+        await _spannerFixture.RunWithTemporaryDatabaseAsync(async databaseId =>
+        {
+            var databaseName = DatabaseName.Format(_spannerFixture.ProjectId, _spannerFixture.InstanceId, databaseId);
+            var databaseToUpdate = databaseAdminClient.GetDatabase(databaseName);
+            var initialDropDbProtection = databaseToUpdate.EnableDropProtection;
+
+            var sample = new UpdateDatabaseAsyncSample();
+            await sample.UpdateDatabaseAsync(_spannerFixture.ProjectId, _spannerFixture.InstanceId, databaseId);
+
+            var updatedDatabase = databaseAdminClient.GetDatabase(databaseName);
+            var updatedDropDbProtection = updatedDatabase.EnableDropProtection;
+
+            Assert.False(initialDropDbProtection);
+            Assert.True(updatedDropDbProtection);
+
+            // Removing database protection after successfull test so that the temporary database can be dropped.
+            await DisableDropDbProtection(databaseAdminClient, updatedDatabase);
+        });
+    }
+
+    private async Task DisableDropDbProtection(DatabaseAdminClient databaseAdminClient, Database database)
```

Comment:
Can't you just create the sample method with a `bool enableDropProtection` and call that here instead of repeating the code?

If you do that, then
- Create the DB with enableDropProtection set to true.
- Test the sample by setting it to false.
- Nothing else to do.